### PR TITLE
Debug drawing circle

### LINKED ISSUE #675: Debug drawing circle
Being able to draw circles in the debug drawing manager. Useful for eg. the brush for painting foliage.

### PR DESCRIPTION
**Pull request by Chainsawkitten**
Adds circles to the debug drawing manager.

Fixes #675

### CODE REVIEW

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 6 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ScriptManager.cpp#L658](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/50b59169540e68aa9efc6c0d899ff4305a0e48a4/src/Engine/Manager/ScriptManager.cpp#L658): Memory leak: value [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3Amemleak)
1. ![MAJOR][MAJOR] [ScriptManager.cpp#L662](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/50b59169540e68aa9efc6c0d899ff4305a0e48a4/src/Engine/Manager/ScriptManager.cpp#L662): Memory leak: value [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3Amemleak)
1. ![MAJOR][MAJOR] [ScriptManager.cpp#L666](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/50b59169540e68aa9efc6c0d899ff4305a0e48a4/src/Engine/Manager/ScriptManager.cpp#L666): Memory leak: value [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3Amemleak)
1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [ScriptManager.cpp#L680](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/50b59169540e68aa9efc6c0d899ff4305a0e48a4/src/Engine/Manager/ScriptManager.cpp#L680): The function 'GetScripts' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L688](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/50b59169540e68aa9efc6c0d899ff4305a0e48a4/src/Engine/Manager/ScriptManager.cpp#L688): The function 'ExecuteScriptMethod' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/Engine/Manager/DebugDrawingManager.cpp [hunk 1]
```diff
@@ -119,6 +131,17 @@ void DebugDrawingManager::Update(float deltaTime) {
...
     }
     
+    // Circles.
+    for (std::size_t i=0; i < circles.size(); ++i) {
+        if (circles[i].duration < 0.f) {
+            circles[i] = circles[circles.size() - 1];
+            circles.pop_back();
+            --i;
+        } else {
+            circles[i].duration -= deltaTime;
```
Comment (Sajdkick) [inline src/Engine/Manager/DebugDrawingManager.cpp]: Are we supposed to use {} after else here?

**Comment by Chainsawkitten** [comment]
Comments fixed. Please reevaluate.

File: src/Video/DebugDrawing.cpp [hunk 2]
```diff
@@ -199,6 +229,19 @@ void DebugDrawing::CreateVertexArray(const glm::vec3* positions, unsigned int po
...
 
+void DebugDrawing::CreateCircle(glm::vec3*& positions, unsigned int& vertexCount, unsigned int detail) {
+    vertexCount = detail * 2;
+    positions = new glm::vec3[vertexCount];
+    
+    unsigned int i = 0;
+    for (unsigned int l = 0; l <= detail; ++l) {
+        float angle = static_cast<float>(l) / detail * 2.0f * glm::pi<float>();
+        positions[i++] = glm::vec3(cos(angle), sin(angle), 0.0f);
+        if (l > 0 && l < detail)
```
Comment (NinjaDanz3r) [inline src/Video/DebugDrawing.cpp]: INFO identifier should not be composed of only 'l' and 'O' [rule](http://localhost:9000/coding_rules#rule_key=vera%2B%2B%3AT010)

File: src/Video/DebugDrawing.cpp [hunk 3]
```diff
@@ -199,6 +229,19 @@ void DebugDrawing::CreateVertexArray(const glm::vec3* positions, unsigned int po
     glBindVertexArray(0);
 }
 
+void DebugDrawing::CreateCircle(glm::vec3*& positions, unsigned int& vertexCount, unsigned int detail) {
+    vertexCount = detail * 2;
+    positions = new glm::vec3[vertexCount];
+    
+    unsigned int i = 0;
+    for (unsigned int l = 0; l <= detail; ++l) {
```
Comment (NinjaDanz3r) [inline src/Video/DebugDrawing.cpp]: INFO identifier should not be composed of only 'l' and 'O' [rule](http://localhost:9000/coding_rules#rule_key=vera%2B%2B%3AT010)

File: src/Video/DebugDrawing.cpp [hunk 4]
```diff
@@ -199,6 +229,19 @@ void DebugDrawing::CreateVertexArray(const glm::vec3* positions, unsigned int po
...
+        float angle = static_cast<float>(l) / detail * 2.0f * glm::pi<float>();
```
Comment (NinjaDanz3r) [inline src/Video/DebugDrawing.cpp]: INFO identifier should not be composed of only 'l' and 'O' [rule](http://localhost:9000/coding_rules#rule_key=vera%2B%2B%3AT010)

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 10 issues
* ![MINOR][MINOR] 7 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### 7 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [ScriptManager.cpp#L63](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/8afb8ffa660320402ca4c3bd0cf95f8001d22760/src/Engine/Manager/ScriptManager.cpp#L63): Variable 'function' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [ScriptManager.cpp#L543](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/8afb8ffa660320402ca4c3bd0cf95f8001d22760/src/Engine/Manager/ScriptManager.cpp#L543): The function 'ClearBreakpoints' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L728](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/8afb8ffa660320402ca4c3bd0cf95f8001d22760/src/Engine/Manager/ScriptManager.cpp#L728): Variable 'json' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [ScriptManager.cpp#L755](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/8afb8ffa660320402ca4c3bd0cf95f8001d22760/src/Engine/Manager/ScriptManager.cpp#L755): The function 'GetStringDeclarationID' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L773](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/8afb8ffa660320402ca4c3bd0cf95f8001d22760/src/Engine/Manager/ScriptManager.cpp#L773): The function 'GetScripts' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L781](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/8afb8ffa660320402ca4c3bd0cf95f8001d22760/src/Engine/Manager/ScriptManager.cpp#L781): The function 'ExecuteScriptMethod' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 7 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [ScriptManager.cpp#L63](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/80d328778713a9dd3b5fb0391d6b2d7b5fa53129/src/Engine/Manager/ScriptManager.cpp#L63): Variable 'function' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [ScriptManager.cpp#L543](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/80d328778713a9dd3b5fb0391d6b2d7b5fa53129/src/Engine/Manager/ScriptManager.cpp#L543): The function 'ClearBreakpoints' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L728](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/80d328778713a9dd3b5fb0391d6b2d7b5fa53129/src/Engine/Manager/ScriptManager.cpp#L728): Variable 'json' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [ScriptManager.cpp#L755](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/80d328778713a9dd3b5fb0391d6b2d7b5fa53129/src/Engine/Manager/ScriptManager.cpp#L755): The function 'GetStringDeclarationID' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L773](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/80d328778713a9dd3b5fb0391d6b2d7b5fa53129/src/Engine/Manager/ScriptManager.cpp#L773): The function 'GetScripts' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L781](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/80d328778713a9dd3b5fb0391d6b2d7b5fa53129/src/Engine/Manager/ScriptManager.cpp#L781): The function 'ExecuteScriptMethod' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 7 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [ScriptManager.cpp#L63](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/6e2f401a5ed705386a888699f9ee3699c5063b3c/src/Engine/Manager/ScriptManager.cpp#L63): Variable 'function' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [ScriptManager.cpp#L543](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/6e2f401a5ed705386a888699f9ee3699c5063b3c/src/Engine/Manager/ScriptManager.cpp#L543): The function 'ClearBreakpoints' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L728](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/6e2f401a5ed705386a888699f9ee3699c5063b3c/src/Engine/Manager/ScriptManager.cpp#L728): Variable 'json' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [ScriptManager.cpp#L755](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/6e2f401a5ed705386a888699f9ee3699c5063b3c/src/Engine/Manager/ScriptManager.cpp#L755): The function 'GetStringDeclarationID' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L773](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/6e2f401a5ed705386a888699f9ee3699c5063b3c/src/Engine/Manager/ScriptManager.cpp#L773): The function 'GetScripts' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L781](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/6e2f401a5ed705386a888699f9ee3699c5063b3c/src/Engine/Manager/ScriptManager.cpp#L781): The function 'ExecuteScriptMethod' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 7 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [ScriptManager.cpp#L63](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/8943d8ebcff0dbd9eb0ccc4f67483c0fa6d7f89e/src/Engine/Manager/ScriptManager.cpp#L63): Variable 'function' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [ScriptManager.cpp#L548](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/8943d8ebcff0dbd9eb0ccc4f67483c0fa6d7f89e/src/Engine/Manager/ScriptManager.cpp#L548): The function 'ClearBreakpoints' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L751](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/8943d8ebcff0dbd9eb0ccc4f67483c0fa6d7f89e/src/Engine/Manager/ScriptManager.cpp#L751): Variable 'json' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [ScriptManager.cpp#L778](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/8943d8ebcff0dbd9eb0ccc4f67483c0fa6d7f89e/src/Engine/Manager/ScriptManager.cpp#L778): The function 'GetStringDeclarationID' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L796](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/8943d8ebcff0dbd9eb0ccc4f67483c0fa6d7f89e/src/Engine/Manager/ScriptManager.cpp#L796): The function 'GetScripts' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L804](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/8943d8ebcff0dbd9eb0ccc4f67483c0fa6d7f89e/src/Engine/Manager/ScriptManager.cpp#L804): The function 'ExecuteScriptMethod' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/Engine/Manager/DebugDrawingManager.hpp [hunk 5]
```diff
@@ -60,6 +60,18 @@ class DebugDrawingManager {
          */
         ENGINE_API void AddPlane(const glm::vec3& position, const glm::vec3& normal, const glm::vec2& size, const glm::vec3& color, float lineWidth = 1.f, float duration = 0.f, bool depthTesting = true);
         
+        /// Add a circle to the world.
+        /**
+         * @param position Center position of the circle.
+         * @param normal Circle normal.
+         * @param radius Radius.
+         * @param color Color of the lines.
+         * @param lineWidth The width of the lines used to draw the plane.
```
Comment (andens) [inline src/Engine/Manager/DebugDrawingManager.hpp]: Not a plane. Also applies to duration.

File: src/Video/DebugDrawing.cpp [hunk 6]
```diff
@@ -156,6 +168,24 @@ void DebugDrawing::DrawPlane(const Plane& plane) {
     glDrawArrays(GL_LINES, 0, 8);
 }
 
+void DebugDrawing::DrawCircle(const Circle& circle) {
+    BindVertexArray(circleVertexArray);
+    
+    glm::mat4 model(glm::scale(glm::mat4(), 0.5f * glm::vec3(circle.radius, circle.radius, circle.radius)));
```
Comment (andens) [inline src/Video/DebugDrawing.cpp]: What's the purpose of halving the radius?

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 7 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [ScriptManager.cpp#L63](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/ef8357ef6b9443499a36c509f1c59d61bd89012e/src/Engine/Manager/ScriptManager.cpp#L63): Variable 'function' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [ScriptManager.cpp#L548](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/ef8357ef6b9443499a36c509f1c59d61bd89012e/src/Engine/Manager/ScriptManager.cpp#L548): The function 'ClearBreakpoints' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L751](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/ef8357ef6b9443499a36c509f1c59d61bd89012e/src/Engine/Manager/ScriptManager.cpp#L751): Variable 'json' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [ScriptManager.cpp#L778](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/ef8357ef6b9443499a36c509f1c59d61bd89012e/src/Engine/Manager/ScriptManager.cpp#L778): The function 'GetStringDeclarationID' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L796](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/ef8357ef6b9443499a36c509f1c59d61bd89012e/src/Engine/Manager/ScriptManager.cpp#L796): The function 'GetScripts' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L804](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/ef8357ef6b9443499a36c509f1c59d61bd89012e/src/Engine/Manager/ScriptManager.cpp#L804): The function 'ExecuteScriptMethod' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 7 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [ScriptManager.cpp#L64](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/e296021f393379391494c1025f53b2f534e12955/src/Engine/Manager/ScriptManager.cpp#L64): Variable 'function' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [ScriptManager.cpp#L564](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/e296021f393379391494c1025f53b2f534e12955/src/Engine/Manager/ScriptManager.cpp#L564): The function 'ClearBreakpoints' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L767](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/e296021f393379391494c1025f53b2f534e12955/src/Engine/Manager/ScriptManager.cpp#L767): Variable 'json' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [ScriptManager.cpp#L794](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/e296021f393379391494c1025f53b2f534e12955/src/Engine/Manager/ScriptManager.cpp#L794): The function 'GetStringDeclarationID' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L812](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/e296021f393379391494c1025f53b2f534e12955/src/Engine/Manager/ScriptManager.cpp#L812): The function 'GetScripts' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L820](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/e296021f393379391494c1025f53b2f534e12955/src/Engine/Manager/ScriptManager.cpp#L820): The function 'ExecuteScriptMethod' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 7 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [ScriptManager.cpp#L64](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/7350124d0e116868f18af154e0cb65603cd49d28/src/Engine/Manager/ScriptManager.cpp#L64): Variable 'function' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [ScriptManager.cpp#L564](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/7350124d0e116868f18af154e0cb65603cd49d28/src/Engine/Manager/ScriptManager.cpp#L564): The function 'ClearBreakpoints' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L767](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/7350124d0e116868f18af154e0cb65603cd49d28/src/Engine/Manager/ScriptManager.cpp#L767): Variable 'json' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [ScriptManager.cpp#L794](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/7350124d0e116868f18af154e0cb65603cd49d28/src/Engine/Manager/ScriptManager.cpp#L794): The function 'GetStringDeclarationID' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L812](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/7350124d0e116868f18af154e0cb65603cd49d28/src/Engine/Manager/ScriptManager.cpp#L812): The function 'GetScripts' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L820](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/7350124d0e116868f18af154e0cb65603cd49d28/src/Engine/Manager/ScriptManager.cpp#L820): The function 'ExecuteScriptMethod' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 7 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [ScriptManager.cpp#L64](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/d0a64ddab3fddcd9bf7c41e1bf3389e48bb44bfa/src/Engine/Manager/ScriptManager.cpp#L64): Variable 'function' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [ScriptManager.cpp#L564](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/d0a64ddab3fddcd9bf7c41e1bf3389e48bb44bfa/src/Engine/Manager/ScriptManager.cpp#L564): The function 'ClearBreakpoints' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L767](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/d0a64ddab3fddcd9bf7c41e1bf3389e48bb44bfa/src/Engine/Manager/ScriptManager.cpp#L767): Variable 'json' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [ScriptManager.cpp#L794](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/d0a64ddab3fddcd9bf7c41e1bf3389e48bb44bfa/src/Engine/Manager/ScriptManager.cpp#L794): The function 'GetStringDeclarationID' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L812](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/d0a64ddab3fddcd9bf7c41e1bf3389e48bb44bfa/src/Engine/Manager/ScriptManager.cpp#L812): The function 'GetScripts' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L820](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/d0a64ddab3fddcd9bf7c41e1bf3389e48bb44bfa/src/Engine/Manager/ScriptManager.cpp#L820): The function 'ExecuteScriptMethod' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 7 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [ScriptManager.cpp#L64](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/b0d722be667253a12392dde570f379f0d5276d6f/src/Engine/Manager/ScriptManager.cpp#L64): Variable 'function' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [ScriptManager.cpp#L564](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/b0d722be667253a12392dde570f379f0d5276d6f/src/Engine/Manager/ScriptManager.cpp#L564): The function 'ClearBreakpoints' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L767](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/b0d722be667253a12392dde570f379f0d5276d6f/src/Engine/Manager/ScriptManager.cpp#L767): Variable 'json' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [ScriptManager.cpp#L794](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/b0d722be667253a12392dde570f379f0d5276d6f/src/Engine/Manager/ScriptManager.cpp#L794): The function 'GetStringDeclarationID' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L812](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/b0d722be667253a12392dde570f379f0d5276d6f/src/Engine/Manager/ScriptManager.cpp#L812): The function 'GetScripts' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L820](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/b0d722be667253a12392dde570f379f0d5276d6f/src/Engine/Manager/ScriptManager.cpp#L820): The function 'ExecuteScriptMethod' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**